src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -44,25 +44,28 @@ internal class GlobalOptionService : IGlobalOptionService
         [SuppressMessage("RoslynDiagnosticsReliability", "RS0033:Importing constructor should be [Obsolete]", Justification = "Used in test code: https://github.com/dotnet/roslyn/issues/42814")]
         public GlobalOptionService(
             [ImportMany] IEnumerable<Lazy<IOptionProvider, LanguageMetadata>> optionProviders,
-            [ImportMany] IEnumerable<Lazy<IOptionPersister>> optionSerializers)
+            [ImportMany] IEnumerable<Lazy<IOptionPersister>> optionPersisters)
         {
             _lazyAllOptions = new Lazy<ImmutableHashSet<IOption>>(() => optionProviders.SelectMany(p => p.Value.Options).ToImmutableHashSet());
-            _optionSerializers = optionSerializers.ToImmutableArray();
-            _serializableOptionsByLanguage = CreateLazySerializableOptionsByLanguage(optionProviders);
+            _optionPersisters = optionPersisters.ToImmutableArray();
+            _serializableOptionsByLanguage = CreateLazySerializableOptionsByLanguage(_optionPersisters, optionProviders);
             _forceComputedLanguages = new HashSet<string>();
             _registeredWorkspaces = ImmutableArray<Workspace>.Empty;
 
             _currentValues = ImmutableDictionary.Create<OptionKey, object?>();
             _changedOptionKeys = ImmutableHashSet<OptionKey>.Empty;
         }
 
-        private static ImmutableDictionary<string, Lazy<ImmutableHashSet<IOption>>> CreateLazySerializableOptionsByLanguage(IEnumerable<Lazy<IOptionProvider, LanguageMetadata>> optionProviders)
+        private static ImmutableDictionary<string, Lazy<ImmutableHashSet<IOption>>> CreateLazySerializableOptionsByLanguage(ImmutableArray<Lazy<IOptionPersister>> optionPersisters, IEnumerable<Lazy<IOptionProvider, LanguageMetadata>> optionProviders)
         {
             var builder = ImmutableDictionary.CreateBuilder<string, Lazy<ImmutableHashSet<IOption>>>();
 
-            foreach (var (language, lazyProvidersAndMetadata) in optionProviders.ToPerLanguageMap())
+            if (!optionPersisters.IsEmpty)
```

Comment:
💭 I wouldn't mind seeing this inverted to just return an empty dictionary before doing the rest of the work.

```csharp
if (optionPersisters.IsEmpty)
{
  return ImmutableDictionary<string, Lazy<ImmutableHashSet<IOption>>>.Empty;
}

var builder = ImmutableDictionary.CreateBuilder<string, Lazy<ImmutableHashSet<IOption>>>();
// ...
```

Reply:
I didn't do that because of the duplication of the horrifically long type name. If we had better inference I'd totally do that, but alas...

Reply:
The type names in this case aren't the same, but it would be interesting to have target-typed generic inference to allow this:

```csharp
return ImmutableDictionary.Create();
```